### PR TITLE
Fix/vexp denormal underflow

### PR DESCRIPTION
**Pull request by dacorvo**
This addresses pathologically long indexing times as reported in #139.

## Why indexing was slow

Attention masks padded columns by adding −1e30 to their scores, then takes a softmax per row:

    w_j = exp(s_j − m) / Σ_k exp(s_k − m),   m = max_k s_k

For a padded column, s_j − m ≈ −1e30, so w_j should be 0.

`vexp` builds 2^f from the float's exponent bits, `(f + 127) << 23`, which is only valid for
f ≥ −126. To stay in range it clamped its input to ln(FLT_MIN) = −87.336, so for every x below
that

    vexp(x) = exp(−87.336) = 2^−126 = FLT_MIN ≈ 1.18e-38

instead of 0. `exp(−87.4)` and `exp(−1e30)` returned the same constant.

The row sum is at least 1 (the max term contributes exp(0)) and in practice much larger, so the
normalized masked weight is

    w_j = FLT_MIN / Σ  <  FLT_MIN

i.e. a subnormal. The scores×V GEMM then computes Σ_j w_j·v_j with a subnormal w_j in every
padded column, and on x86 each such FMA takes the denormal microcode assist (~100× the cost of
a normal one). In a ragged batch, padding is most of the columns, so most of the GEMM ran on the
slow path.

## Fix

Return exactly 0 for x < ln(FLT_MIN). The true exp(x) is below the smallest normal float there,
so it contributes nothing representable to the row sum (≤ cols·FLT_MIN ≈ 6e-36 relative), and
the masked weight becomes the 0 it should be. For x ≥ ln(FLT_MIN) the expression is unchanged,
so in-range results are bit-identical.

Ragged batch (Xeon 8275CL, bench_backends 16×mixed): embed 5880 → 1137 ms, rerank 11186 → 2525 ms.

### CODE REVIEW

File: src/inference/blas.rs [hunk 1]
```diff
@@ -156,12 +157,20 @@ mod seam {
         const LN2_LO: f32 = -2.121_944_4e-4;
         const MAGIC: f32 = 12582912.0; // 1.5·2^23: add then subtract rounds to nearest integer
         const C: [f32; 5] = [1.0 / 720.0, 1.0 / 120.0, 1.0 / 24.0, 1.0 / 6.0, 0.5];
+        // ln(FLT_MIN): the exponent trick below wraps past it, and e^x is subnormal there anyway.
+        const UNDERFLOW: f32 = -87.336;
```
[the remark this answers is not in the document]
Comment (dacorvo) [reply src/inference/blas.rs]: Bug is real, fix is not.
